### PR TITLE
test: Add more unit tests [DEV-2506]

### PR DESCRIPTION
**Pull request by anikitinDSR**

### CODE REVIEW

**Comment by ankurdotb** [comment]
Task linked: [DEV-2506 Create unit tests for DID URL dereferencing](https://app.clickup.com/t/6600954/DEV-2506)

File: tests/unit/diddoc/request/diddoc_service_params_test.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+		"Positive. VersionTime + Service case + relativeRef",
+		QueriesDIDDocTestCase{
+			didURL:             fmt.Sprintf("/1.0/identifiers/%s?versionTime=%s&service=%s&relativeRef=foo", testconstants.ValidDid, testconstants.CreatedAfter.Format(time.RFC3339), testconstants.ValidServiceId),
+			resolutionType:     types.DIDJSONLD,
+			expectedResolution: types.NewServiceResult(testconstants.ValidService.ServiceEndpoint[0] + "foo"),
+			expectedError:      nil,
+		},
+	),
+
+	// Negative Cases
```
Comment (abdulla-ashurov) [inline tests/unit/diddoc/request/diddoc_service_params_test.go:89]: Let's add a negative test for testing `relativeRef` query without `service` query.

File: tests/unit/test_utils.go [hunk 2]
```diff
@@ -83,27 +94,41 @@ func (ls MockLedgerService) QueryDIDDoc(did string, version string) (*didTypes.D
...
 
 // TODO: add unit tests for testing QueryCollectionResources method.
 func (ls MockLedgerService) QueryCollectionResources(did string) ([]*resourceTypes.Metadata, *types.IdentityError) {
-	if ls.Did.Id != did || ls.Resource.Metadata == nil {
+	if ls.Did.Id != did {
 		return []*resourceTypes.Metadata{}, types.NewNotFoundError(did, types.JSON, nil, true)
 	}
 
-	return []*resourceTypes.Metadata{ls.Resource.Metadata}, nil
+	var metadata_list = make([]*resourceTypes.Metadata, len(ls.Resources))
```
Comment (abdulla-ashurov) [inline tests/unit/test_utils.go:126]: Rename this local variable from `metadata_list` to `metadataList` to match golang code style.

**Review by abdulla-ashurov — CHANGES_REQUESTED** [review]
A great job! 🔥

**Review by abdulla-ashurov — APPROVED** [review]

**Pull request merged**